### DISCUSSION TITLE
Can't seem to figure out state type. Need help

### DISCUSSION THREAD

**Discussion by omarcusmoreira** [rematch/rematch#965] [Q&A]
Hi, people from the internet.

I am trying to fetch some data from github public api and I created a store and user model for that.
I can successfully fetch the data but it loads in my state with a "data" propriety before de UserData type that I have created.

So when I access the state.User I see the userdata from github but in order to use the name, I need to use state.User.data.name and TS doesn't agree to that. It gives me the error saying that state doesn't have the "data" propriety (of course it doesn't) but the Browser shows the actual data (so I know I am fetching the right thing.

I think I don't know how to setup the state type properly.

**Comment by semoal** [comment]
You have a guide here: https://rematchjs.org/docs/getting-started/typescript
Also you have a complete example in the Github repository. 

Feel free to provide a minimal example and we could help a bit more
